src/util/colorScale.js

```diff
@@ -45,6 +46,46 @@ const getDiscreteValuesFromTree = (nodes, nodesToo, attr) => {
   return domain;
 };
 
+/**
+ * Dynamically create legend values based on visibility for ordinal and categorical scale types.
+ */
+export const createVisibleLegendValues = (
+  {colorBy, type, legendValues, tree, treeToo = undefined, controls = undefined}
+) => {
+  let visibility = tree.visibility;
+  let visibilityToo = treeToo ? treeToo.visibility : undefined;
+
+  // Create visibility if undefined.
+  if (!visibility || !visibilityToo) {
```

Comment:
For single trees, `visibilityToo` is falsey and so this if block is run, which isn't what you want!